Added Kogan Aircons and Fans. Fixed Mirabella Genio Powerboard from 4 outlets to 5.

sambarlick opened the pull request:
Added Kogan Aircons and Fans. Fixed Mirabella Genio Powerboard from 4 switches to 5 (4 outlets + 1 USB switch).

make-all commented on custom_components/tuya_local/devices/kogan_kawfpac16pya_airconditioner.yaml:
```diff
@@ -0,0 +1,115 @@
+            - dps_val: DRY
+              icon: "mdi:water"
+              value: dry
+            - dps_val: Heat
```
Both these configs seem the same as fersk_vind_2_climate.yaml differing only by this "HEAT/Heat/missing" mode, which is strange.

make-all commented on custom_components/tuya_local/devices/kogan_kablfan1csa_fan.yaml:
```diff
@@ -0,0 +1,109 @@
+name: Kogan Smarterhome Bladeless Purifier Fan
+products:
+  - id: kablfan1csa
```
Are these product ids actually matching the catalogue part numbers?

The id here should be from the Tuya protocol (broadcast as part of the discovery protocol, as reported by tinytuya, or from the device info related cloud API functions (generally the same functions that can tell you the local key).

make-all commented on custom_components/tuya_local/devices/kogan_kawfpac12ya_airconditioner.yaml:
```diff
@@ -0,0 +1,115 @@
+
+    - id: 5
+      name: fan_mode
+      type: integer
```
Assuming strings

make-all commented:
I'll merge this as-is, but there are still some possible improvements that depend on answers to the questions above that are still open:
- ranges of on-device timers to allow them to be exposed as number entities
- product ids (for future matching, these have to correspond to the Tuya discovery ids)

sambarlick commented:
Hi Jason,

I'm sorry for not getting back to you sooner.
I've been really busy with work. I have some time this weekend to go
through all the emails you've sent.

Apologies once again,
Sam

On Thu, 8 Feb 2024, 16:15 Jason Rumney, ***@***.***> wrote:

> Merged #1480 <https://github.com/make-all/tuya-local/pull/1480> into main.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/make-all/tuya-local/pull/1480#event-11740869751>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALKDPDBU26EZIDRJQCNHHT3YSRNQRAVCNFSM6AAAAABBPKCUSWVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJRG42DAOBWHE3TKMI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

sambarlick commented:
Hi Jason,

Thank you for taking the time to send those emails. Again I'm sorry for the
lack of response.

Firstly, the difference between the 2 aircons pac12ya and pac16pya is
simply the heating mode. 12ya doesn't have it.

Secondly, I expanded on the temperature unit functionality you added. I can
confirm that it changes the unit and range on the device itself but doesn't
update in Home Assistant (see attached screenshot).

I don't know why but combining the 2 fans, fan1csa and fansa "broke" the
devices. The became greyed out in Home Assistant. I re-added the seperated
yamls and they both worked again.

The difference between these 2 fans is that the sound dp is different and
one has a minute countdown dp and the other doesn't.

I have attached a screenshot from tuya developer IoT website of the device
information for the fan1csa device. Can I provide anymore info from my tuya
IoT account?

I accept that there are conventions and I apologise for my ignorance of
these. I used the dp values that I found in localtuya and matched them with
what I saw on the tuya IoT developer site.

I changed the mode for the fans that I had as boost to strong and it didn't
work. So I changed it back to boost. I will try this again.

I'm sure I have missed things... Please let me know.

Also, some things I'm unsure of;
   - is it possible to add icons to the fan modes?
     When I associated icons with the modes it         changed the device
icon depending on the         mode. (See attached screenshot)
   - sometimes the changes I make in HA don't.       happen on the devices.
Alternatively                     sometimes changes made on the device
       don't reflect in Home Assistant. Is there               something I
have not done correctly?

Thanks,
Sam

On Thu, 8 Feb 2024, 16:15 Jason Rumney, ***@***.***> wrote:

> Merged #1480 <https://github.com/make-all/tuya-local/pull/1480> into main.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/make-all/tuya-local/pull/1480#event-11740869751>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALKDPDBU26EZIDRJQCNHHT3YSRNQRAVCNFSM6AAAAABBPKCUSWVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJRG42DAOBWHE3TKMI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

make-all commented:
Due to a better way of defining state dependant icons introduced in HA 2024.2, I don't want to add any more complex icon logic at this stage, as it will need to be redone when the new support is added (the new support will likely be shared across devices, so much easier than having to define it in every config).

For other issues, it may help to look for log messages (you may need to temporarily switch debug logging on, which you can do from the Settings / Devices page) or the device diagnostics.

make-all commented:
> I don't know why but combining the 2 fans, fan1csa and fansa "broke" the devices. The became greyed out in Home Assistant. I re-added the seperated yamls and they both worked again.

You need to remove the old devices and re-add them, because the config file name has changed. They will become greyed out because the config file named in the config under the original names you made are missing, replaced by "kogan_bladeless_fan.yml"
